src/Acquisition/tests.py

```diff
@@ -1620,25 +1655,31 @@ def test_cant_persist_acquisition_wrappers_classic():
     >>> pickler = cPickle.Pickler(file, 1)
 
     >>> def persistent_id(obj):
+    ...     if not hasattr(obj, '_p_oid'): return None
```

Comment:
Exactly. The `cPickle` implementation used in CPython 2.x  and 3.x's C-accelerated implementation simply invokes `PyObject_Call` on the user-provided function and checks its return value; it *does not* check for exceptions being raised (as far as I can tell; I'm not a C API expert), so they are silently ignored. In contrast, PyPy (or the non-C-accelerated `pickle` module) make the call at the Python level, so exceptions that the `persistent_id` function raise are propagated all the way up, breaking this test. Hence the need to keep `persistent_id` from raising when called with a "non-persistent" object.